admin/src/js/services/format-date.js

```diff
@@ -42,79 +41,10 @@ const moment = require('moment');
         return moment(date).format(config[key]);
       };
 
-      const getDateDiff = function(date, options) {
-        let end = options.end ? moment(options.end) : moment(); // default to now
-        end = end.startOf('day'); // remove the time component
-        for (let i = 0; i < config.ageBreaks.length; i++) {
-          const ageBreak = config.ageBreaks[i];
-          const diff = date.diff(end, ageBreak.unit);
-          if (Math.abs(diff) > ageBreak.min) {
-            return { quantity: diff, key: ageBreak.key };
-          }
-        }
-        return { quantity: 0, key: { singular: 'd', plural: 'dd' } };
-      };
-
-      const getTaskDueDate = function(given) {
-        const date = moment(given).startOf('day');
-        const today = moment().startOf('day');
-        const diff = date.diff(today, 'days');
-        if (diff <= 0) {
-          return $translate.instant('task.overdue');
-        }
-        if (diff <= config.taskDayLimit) {
-          return $translate.instant('task.days.left', { DAYS: diff }, 'messageformat');
-        }
-        return '';
-      };
-
-      const relativeDate = function(date, options) {
-        const diff = getDateDiff(moment(date).startOf('day'), options);
-        if (options.humanize) {
-          if (diff.quantity === 0) {
-            return $translate.instant('today');
-          }
-          if (diff.quantity === 1) {
-            return $translate.instant('tomorrow');
-          }
-          if (diff.quantity === -1) {
-            return $translate.instant('yesterday');
-          }
-        }
-        const quantity = Math.abs(diff.quantity);
-        const key = quantity === 1 ? diff.key.singular : diff.key.plural;
-        const output = MomentLocaleData().relativeTime(quantity, true, key);
-        if (options.suffix) {
-          return MomentLocaleData().pastFuture(diff.quantity, output);
-        }
-        return output;
-      };
-
       return {
-        date: function(date) {
-          return format(date, 'date');
-        },
         datetime: function(date) {
           return format(date, 'datetime');
         },
-        relative: function(date, options) {
-          options = options || {};
-          if (options.task) {
-            return getTaskDueDate(date);
-          }
-          if (options.withoutTime) {
-            return relativeDate(date, { suffix: true, humanize: true });
-          }
-          return moment(date).fromNow();
-        },
-        age: function(date, options) {
-          options = options || {};
-          return relativeDate(date, options);
-        },
-        time: function(date) {
-          return format(date, 'time');
-        }
```

Comment:
This is amazing work, thank you! I love to see admin slowly being slimmed down.

Now that you've eliminated almost all of this code I think we can remove the factory altogether. It looks like it's only used in one place, in the EditUser controller. The only place it's called we're already getting the `settings` so we can just use that and format the date inline or in a private function. This is how the MessageQueue controller works so you can use that as an example.